src/sentry/models/group.py

```diff
@@ -478,17 +478,35 @@ def get_oldest_event_for_environments(self, environments=()):
             project_id=self.project_id,
         )
 
+    def __get_release(self, project_id, group_id, first=True):
+        from sentry.models import GroupRelease, Release
+
+        orderby = "first_seen" if first else "-last_seen"
+        group_releases = list(
+            GroupRelease.objects.filter(group_id=group_id, project_id=project_id,).order_by(
+                orderby
+            )[:1]
+        )
+        if group_releases:
+            try:
+                release = Release.objects.get(id=group_releases[0].release_id)
+                return release.version
+            except Release.DoesNotExist:
+                return None
```

Comment:
To fit this into one query, maybe we could just not convert `group_releases` to a list and use it as a subquery instead. So we could use 
```
Release.objects.get(
    id__in=GroupRelease.objects.filter(group_id=group_id, project_id=project_id,).order_by(orderby).values("release_id")[:1]
)
```

Then we can just catch `Release.DoesNotExist` and skip the extra checks as well. Probably worth just testing this out and posting the sql here so that we can verify it makes sense.

Reply:
This is great Dan.